src/utils/__tests__/internalLinks-test.js

```diff
@@ -37,6 +37,100 @@ describe('isInternalLink', () => {
     [false, 'same domain, double slash', 'https://example.com//#narrow/stream/jest'],
     [false, 'same domain, triple slash', 'https://example.com///#narrow/stream/jest'],
 
+    // These examples may be odd-looking URLs, but are nevertheless valid.
+    [true, 'scheme-relative', '//example.com/#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '.#narrow/stream/jest'],
+    [true, 'path-relative, nop path', './#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '..#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '../#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/..#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/../#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/bar/../../../baz/.././#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/.#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/./#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/..#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/../#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/..#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/../#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/bar/../../../baz/.././#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/.#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/./#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/..#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/../#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/foo/..#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/foo/../#narrow/stream/jest'],
+    [
+      true,
+      'same domain, nop path',
+      'https://example.com/foo/bar/../../../baz/.././#narrow/stream/jest',
+    ],
+    [true, 'same domain, %-encoded host', 'https://%65xample%2ecom/#narrow/stream/jest'],
+    [
+      true,
+      'same domain, punycoded host',
+      'https://example.xn--h2brj9c/#narrow/stream/jest',
+      new URL('https://example.भारत/'),
+    ],
+    [
+      true,
+      'same domain, punycodable host',
+      'https://example.भारत/#narrow/stream/jest',
+      new URL('https://example.भारत/'),
+    ],
+    // This one fails because our polyfilled URL implementation has IDNA stripped out.
+    // [true, 'same domain, IDNA-mappable', 'https://ℯⅩªm🄿ₗℰ.ℭᴼⓂ/'], // FAILS
```

Comment:
Nit: `https://ℯⅩªm🄿ₗℰ.ℭᴼⓂ/#narrow/stream/jest` (e.g.), I think.

Also, I think we're not using the polyfilled implementation in these tests; we don't (e.g.) have a line like `import 'react-native-url-polyfill/auto';` here or in our Jest setup.